OpenRA.Game/WorldUtils.cs

```diff
@@ -73,10 +53,15 @@ public static void PlayVoiceForOrders(this World w, Order[] orders)
 				if (o.Subject.Destroyed)
 					continue;
 
+				var hasVoice = o.Subject.TraitsImplementing<IVoiced>();
+				if (hasVoice == null)
```

Comment:
Suggestion: rename `hasVoice` to `voices` here and use it instead of doing another `o.Subject.TraitsImplementing<IVoiced>()` in the `foreach` 3 lines below.